arrow-cast/src/cast.rs

```diff
@@ -1955,12 +1956,26 @@ fn cast_decimal_to_decimal_safe<const BYTE_WIDTH1: usize, const BYTE_WIDTH2: usi
         // For example, input_scale is 4 and output_scale is 3;
         // Original value is 11234_i128, and will be cast to 1123_i128.
         let div = 10_i128.pow((input_scale - output_scale) as u32);
+        let half = div / 2;
+        let neg_half = half.neg();
```

Comment:
```suggestion
        let neg_half = half.wrapping_neg();
```
As we've divided by 2 this can't overflow